src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -666,6 +666,11 @@ internal sealed override void AfterAddingTypeMembersChecks(ConversionsBase conve
 
             ParameterHelpers.EnsureIsReadOnlyAttributeExists(compilation, Parameters, diagnostics, modifyCompilation: true);
 
+            if (ReturnType.ContainsNativeInteger())
+            {
+                compilation.EnsureNativeIntegerAttributeExists(diagnostics, ReturnTypeSyntax.Location, modifyCompilation: true);
```

Comment:
>EnsureNativeIntegerAttributeExists [](start = 28, length = 34)

Where do we synthesize the attribute for this symbol?

Reply:
`MethodSymbol.AddSynthesizedReturnTypeAttributes()`

---
In reply to: [372664832](https://github.com/dotnet/roslyn/pull/41160#discussion_r372664832) [](ancestors = 372664832)

Reply:
> MethodSymbol.AddSynthesizedReturnTypeAttributes()

Would it make sense to consolidate checks like this in the base type as well? So that the checks and the code that synthesizes the attributes is next to each other.

---
In reply to: [372761155](https://github.com/dotnet/roslyn/pull/41160#discussion_r372761155) [](ancestors = 372761155,372664832)